### DISCUSSION TITLE
[Backend] graphlib, ngraph or graphology?

### DISCUSSION THREAD

**Discussion by dleeftink** [SkepticMystic/graph-analysis#44] [General]
Seeing how graphlib isn't actively maintained anymore, I was wondering why this library was chosen over alternatives as [ngraph](https://github.com/anvaka/ngraph) or [graphology](https://graphology.github.io/). The latter come packed with plugins and algorithms (clustering, traversal), with graphology's implementation strictly typed and fully documented. I wonder what graphlib provides compared to these libraries?

**Comment by SkepticMystic** [comment]
Haha graphlib has nothing on these other libs 😂 
We can definitely look into using a better one, I see no reason why not. We should discuss which would be best, and just take some time to refactor everything.
@HEmile, what do you think?

**Comment by SkepticMystic** [comment]
After a quick look, I think graphology is better

**Reply by HEmile** [reply]
I think I agree, it has a nice standard library with the Louvain algorithm we wanted to implement anyways :)

**Comment by SkepticMystic** [comment]
@dleeftink the refactor was actually pretty simple! We've moved over to graphology already, and that allowed us to very quickly add Louvain community detection! So it'll be up in the next release :)

**Reply by dleeftink** [reply]
That was quick indeed. Hopefully it'll be accustoming to incoming functionality, graphology seems quite extendable.
